sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -1466,9 +1468,24 @@ <T> Mono<FeedResponse<T>> readMany(
 
         requestOptions.setMaxDegreeOfParallelism(-1);
 
+        CosmosAsyncClient client = this.getDatabase().getClient();
+        CosmosPagedFluxOptions fluxOptions = new CosmosPagedFluxOptions();
+        fluxOptions.setMaxItemCount(itemIdentityList != null ? itemIdentityList.size() : 0);
```

Comment:
We are setting the page size to flux options, whereas when executing readMany, we get the page size using `CosmosQueryRequestOptions` in `PipelinedQueryExecutionContext`
Though this will not change the existing behavior, but I doubt if it will fix the page size.

Reply:
QueryFeedOperationState is responsible for populating the maxItemCount from PagedFluxOptions to QueryRequestOptions. This works as intended.